src/client/plugin/lib.ts

```diff
@@ -15,6 +15,19 @@ export function parseCookies(cookieStr: string) {
     )
 }
 
+export function parseJwt(token: string) {
+  const base64Url = token.split(".")[1]
+  const base64 = base64Url.replace(/-/g, "+").replace(/_/g, "/")
+  const jsonPayload = decodeURIComponent(
+    atob(base64)
+      .split("")
+      .map((c) => "%" + ("00" + c.charCodeAt(0).toString(16)).slice(-2))
+      .join(""),
+  )
```

Comment:
There is a lot going on here that isn't clear to me. Could you add a comment or link?

Reply:
Thank you. Could you add a unit test for this function?

Reply:
Hey @n4bb12, apologies for the delay, I was on vacation for the past week. I just added a unit test if you could take another look :)